From 58408f384e7816f126e1cb6d7d9153cc635c95f6 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-58408f384e7-20211217T000703Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sat, 6 Nov 2021 15:04:34 +0100
Subject: [PATCH] Makefile: don't invoke msgfmt with --statistics
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove the --statistics flag that I added in 5e9637c6297 (i18n: add
infrastructure for translating Git with gettext, 2011-11-18). Our
Makefile output is good about reducing verbosity by default, except in
this case:

    $ rm -rf po/build/locale/e*; time make -j $(nproc) all
        SUBDIR templates
        MKDIR -p po/build/locale/el/LC_MESSAGES
        MSGFMT po/build/locale/el/LC_MESSAGES/git.mo
        MKDIR -p po/build/locale/es/LC_MESSAGES
        MSGFMT po/build/locale/es/LC_MESSAGES/git.mo
    1038 translated messages, 3325 untranslated messages.
    5230 translated messages.

I didn't have any good reason for using --statistics at the time other
than ad-hoc eyeballing of the output. We don't need to spew out
exactly how many messages we've got translated every time. Now we'll
instead emit:

    $ rm -rf po/build/locale/e*; time make -j $(nproc) all
        SUBDIR templates
        MKDIR -p po/build/locale/el/LC_MESSAGES
        MSGFMT po/build/locale/el/LC_MESSAGES/git.mo
        MKDIR -p po/build/locale/es/LC_MESSAGES
        MSGFMT po/build/locale/es/LC_MESSAGES/git.mo

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
Range-diff:
 1:  b763e74f000 <  -:  ----------- Makefile: move git-SCRIPT-DEFINES adjacent to $(SCRIPT_DEFINES)
 2:  4bb75d22d97 <  -:  ----------- Makefile: remove $(GIT_VERSION) from $(SCRIPT_DEFINES)
 3:  3bbd5cbf0f4 <  -:  ----------- Makefile: remove $(NO_CURL) from $(SCRIPT_DEFINES)
 4:  d50bd32cb82 <  -:  ----------- git-instaweb: unconditionally assume that gitweb is mod_perl capable
 5:  f15f64faf9f <  -:  ----------- git-sh-setup: remove unused sane_egrep() function
 6:  974d734cb8e <  -:  ----------- git-sh-setup: remove "sane_grep", it's not needed anymore
 7:  1621ca72c1d =  1:  58408f384e7 Makefile: don't invoke msgfmt with --statistics
 8:  b7c36c9fea0 <  -:  ----------- Makefile: don't set up "perl/build" rules under NO_PERL=Y
 9:  29b000eb0f1 <  -:  ----------- Makefile: use "=" not ":=" for po/* and perl/*
10:  daead5ec293 <  -:  ----------- Makefile: clean perl/build/ even with NO_PERL=Y
11:  3c987590740 <  -:  ----------- Makefile: remove "mv $@ $@+" dance redundant to .DELETE_ON_ERROR
12:  b57f582ccd3 <  -:  ----------- Makefile: guard Perl-only variable assignments
13:  fcdee92f64c <  -:  ----------- Makefile: change "ifndef NO_PERL" to "ifdef NO_PERL"
14:  1e25b532ca2 <  -:  ----------- Makefile: adjust Perl-related comments & whitespace
15:  77d9855bfcf <  -:  ----------- Makefile: correct "GIT-PERL-{DEFINES,HEADER}" dependency graph
16:  6004cdcd8d9 <  -:  ----------- Makefile: create a GIT-PYTHON-DEFINES, like "PERL"
17:  17b30e96057 <  -:  ----------- Makefile: stop needing @@GIT_VERSION@@ in *.perl scripts
18:  30ddf7da2c8 <  -:  ----------- Makefiles: add "shared.mak", move ".DELETE_ON_ERROR" to it
19:  f378a7dc35e <  -:  ----------- Makefile: move $(comma), $(empty) and $(space) to shared.mak
20:  13cbb851d32 <  -:  ----------- Makefile: re-add and use the "shellquote" macros
21:  337953e4994 <  -:  ----------- Makefile: add a "TRACK_template" for GIT-*{FLAGS,DEFINES,...}
22:  5bb597c1993 <  -:  ----------- Makefile: add "$(QUIET)" boilerplate to shared.mak
23:  3c4d0589667 <  -:  ----------- Makefile: use $(wspfx) for $(QUIET...) in shared.mak
24:  be5882b2c99 <  -:  ----------- Makefiles: add and use wildcard "mkdir -p" template
25:  2710f8af6cd <  -:  ----------- Makefile: correct the dependency graph of hook-list.h
26:  59f22a0269a <  -:  ----------- Makefile: use $(file) I/O instead of "FORCE" when possible
27:  dd569a59c74 <  -:  ----------- Makefile: disable GNU make built-in wildcard rules
28:  4168a7e3b30 <  -:  ----------- Makefile: define $(LIB_H) in terms of $(FIND_SOURCE_FILES)
29:  48a3927d972 <  -:  ----------- Makefile: move ".SUFFIXES" rule to shared.mak

 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9c00a793e47..b9d0f063eff 100644
--- a/Makefile
+++ b/Makefile
@@ -1880,7 +1880,7 @@ ifdef GIT_TEST_CMP_USE_COPIED_CONTEXT
 endif
 
 ifndef NO_MSGFMT_EXTENDED_OPTIONS
-	MSGFMT += --check --statistics
+	MSGFMT += --check
 endif
 
 ifdef HAVE_CLOCK_GETTIME
-- 
2.34.1.1119.g7a3fc8778ee

